planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp

```diff
@@ -1916,33 +1920,42 @@ void AvoidanceModule::modifyPathVelocityToPreventAccelerationOnAvoidance(Shifted
     return;
   }
 
-  // calc time to the shift-change point
   constexpr auto NO_ACCEL_TIME_THR = 3.0;
-  const auto s = avoidance_data_.arclength_from_ego.at(target_idx) -
-                 avoidance_data_.arclength_from_ego.at(ego_idx);
-  const auto t = s / std::max(getEgoSpeed(), 1.0);
-  if (t > NO_ACCEL_TIME_THR) {
-    DEBUG_PRINT(
-      "shift point is far (s: %f, t: %f, ego_i: %lu, target_i: %lu). No velocity limit is applied.",
-      s, t, ego_idx, target_idx);
-    return;
-  }
 
-  // calc max velocity with given acceleration
-  const auto v0 = getEgoSpeed();
-  const auto vmax = std::max(
-    parameters_.min_avoidance_speed_for_acc_prevention,
-    std::sqrt(v0 * v0 + 2.0 * s * parameters_.max_avoidance_acceleration));
+  // update ego velocity if the shift point is far
+  const auto s_from_ego = avoidance_data_.arclength_from_ego.at(target_idx) -
+                          avoidance_data_.arclength_from_ego.at(ego_idx);
+  const auto t_from_ego = s_from_ego / std::max(getEgoSpeed(), 1.0);
+  if (t_from_ego > NO_ACCEL_TIME_THR) {
+    *ego_velocity_starting_avoidance_ptr_ = getEgoSpeed();
+  }
+
+  // calc index and velocity to NO_ACCEL_TIME_THR
+  const auto v0 = *ego_velocity_starting_avoidance_ptr_;
+  auto vmax = 0.0;
+  size_t insert_idx = ego_idx;
+  for (size_t i = ego_idx; i < target_idx; ++i) {
+    const auto s =
+      avoidance_data_.arclength_from_ego.at(target_idx) - avoidance_data_.arclength_from_ego.at(i);
+    const auto t = s / std::max(v0, 1.0);
+    if (t < NO_ACCEL_TIME_THR) {
```

Comment:
Will this if statement always be executed in any case?

(Is there a case where vmax remains at the initial value of 0?)

Reply:
`s` is decreasing with increasing `i`.
However, there is no guarantee to become `t < NO_ACCEL_TIME_THR`.
It should handle the initial value.
I'll fix it.

Reply:
Fixed.
https://github.com/autowarefoundation/autoware.universe/pull/909/commits/667e1c25c724a7c62f7851a2cf75b0110d57b67b